orchestration/dagster_orchestration/hca_orchestration/contrib/data_repo.py

```diff
@@ -0,0 +1,58 @@
+"""
```

Comment:
As the comment says, this is an attempt to abstract over the raw jade data repo client and provide some insulation and testability. Importantly, all operations here automatically poll on jobs rather than relying on the consuming code to handle that.